packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -612,163 +612,80 @@ contract OPContractsManagerUpgrader is OPContractsManagerBase {
 
         // If the SuperchainConfig is not already upgraded, upgrade it. NOTE that this type of
         // upgrade means that chains can ONLY be upgraded via this OPCM contract if they use the
-        // same SuperchainConfig contract. We will assert this later.
+        // same SuperchainConfig contract. This line of code will ONLY get hit if the
+        // SuperchainConfig was actually modified as part of this upgrade.
         if (_superchainProxyAdmin.getProxyImplementation(address(_superchainConfig)) != impls.superchainConfigImpl) {
             // Attempt to upgrade. If the ProxyAdmin is not the SuperchainConfig's admin, this will revert.
-            upgradeToAndCall(
-                _superchainProxyAdmin,
-                address(_superchainConfig),
-                impls.superchainConfigImpl,
-                abi.encodeCall(ISuperchainConfig.upgrade, ())
-            );
+            upgradeTo(_superchainProxyAdmin, address(_superchainConfig), impls.superchainConfigImpl);
         }
 
         // Loop through each chain and upgrade.
         for (uint256 i = 0; i < _opChainConfigs.length; i++) {
             assertValidOpChainConfig(_opChainConfigs[i]);
 
+            // Make sure the SuperchainConfig is the same as the shared one.
+            if (_superchainConfig != _opChainConfigs[i].systemConfigProxy.superchainConfig()) {
+                revert OPContractsManagerUpgrader_SuperchainConfigMismatch();
+            }
```

Comment:
hmmmmm... we need to get clear on the requirement here. 

This might be a good time to implement #16359.

@AmadiMichael will also need to be aware that this PR will cause conflicts with his current work to fix the superchain config upgrade path. FWIW I think that's OK.

Reply:
I vote we just include this PR as-is and fix the upgrade thing in another PR to avoid stuffing too many changes here

Reply:
Right, I agree that we should keep them separate. 

When I left this comment, I thought this was a net new check, but I see that it was just moved from elswhere.

Good to keep this for now.